ctf2ctf.cpp

```diff
@@ -1335,15 +1335,32 @@ struct Event
 
         cpuId = rawCpuId.value();
 
-        tid = context->tid(cpuId);
-        pid = context->pid(tid);
+        if (context->options.ctxPidTid) {
+            const bt_definition* context_scope = bt_ctf_get_top_level_scope(event, BT_STREAM_EVENT_CONTEXT);
+            auto o_pid = get_int64(event, context_scope, "vpid");
+            if (o_pid) {
+                pid = o_pid.value();
+                context->setTid(cpuId, pid);
+            }
+            auto o_tid = get_int64(event, context_scope, "vtid");
+            int64_t r_tid = -1;
+            if (o_tid) {
+                r_tid = o_tid.value();
+                context->setPid(r_tid, pid);
+            }
+            tid = r_tid - pid;
```

Comment:
can you please explain a bit what you are doing here? I don't understand why you are subtracting the pid from the r_tid to build the tid?

please add a short comment that explains what the actual values are ew get from vpid/vtid, and how they relate to the real pid/tid we get from kernel trace points

Reply:
I subtract pid from tid to have nicer representaion of app's main thread, as it will have id of 0, otherwise thread ids are crazy high numbers.

Reply:
well yes, but that's life - PID/TID values are not under our control and we should show them as-is. If you want to have such "relative" numbers, please add that in a separate commit and make it optional - this does not seem to be related to the actual task you are solving here (using vpid/vtid)